File: 3731.find-missing-elements/solution.kt
class Solution {
    fun findMissingElements(nums: IntArray): List<Int> {
        nums.distinct()
        nums.sort()

        val res = mutableListOf<Int>()
        var expected = nums.first()
        
        for (n in nums) {
            res += expected until n
            expected = n + 1
        }

        return res
    }
}

File: 3731.find-missing-elements/solution.go
func findMissingElements(nums []int) []int {
    set := make(map[int]bool)
    min := nums[0]
    max := nums[0]

    for _, n := range nums {
        set[n] = true
        if n < min {
            min = n
        }
        if n > max {
            max = n
        }
    }

    var res []int

    for i := min; i < max; i++ {
        if !set[i] {
            res = append(res, i)
        }
    }

    return res
}